Client/Frontend/Settings/LoginDetailViewController.swift

```diff
@@ -178,4 +250,52 @@ extension LoginDetailViewController {
         tableView.editing = false
         navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .Edit, target: self, action: "SELedit")
     }
+
+    func SELwillShowMenuController() {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: UIMenuControllerWillShowMenuNotification, object: nil)
+
+        let menuController = UIMenuController.sharedMenuController()
+        guard let cell = menuControllerCell,
+              let textSize = cell.descriptionTextSize else {
+            return
+        }
+
+        // The description label constraints are such that it extends full width of the cell instead of only 
+        // the size of its text. The reason is because when the description is used as a password, the dots
+        // are slightly larger characters than the font size which causes the password text to be truncated 
+        // even though the revealed text fits. Since the label is actually full width, the menu controller will
+        // display in its center by default which looks weird with small passwords. To prevent this,
+        // the actual size of the text is used to determine where to correctly place the menu.
+
+        var descriptionFrame = cell.descriptionLabel.frame
+        descriptionFrame.size = textSize
+
+        menuController.arrowDirection = .Up
+        menuController.setTargetRect(descriptionFrame, inView: cell)
+        menuController.setMenuVisible(true, animated: true)
+    }
+
+    func SELwillHideMenuController() {
+        menuControllerCell = nil
+
+        // Re-add observer
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "SELwillShowMenuController", name: UIMenuControllerWillShowMenuNotification, object: nil)
+    }
+}
+
+// MARK: - Cell Delegate
+extension LoginDetailViewController: LoginTableViewCellDelegate {
+
+    func didSelectOpenAndFillForCell(cell: LoginTableViewCell) {
+        navigationController?.dismissViewControllerAnimated(true, completion: {
+            if let appDelegate = UIApplication.sharedApplication().delegate as? AppDelegate {
+                let rootNavigationController = appDelegate.rootViewController
+                rootNavigationController.popViewControllerAnimated(true)
+                if let formSubmitURL = self.login.formSubmitURL,
+                   let url = NSURL(string: formSubmitURL) {
+                    appDelegate.browserViewController.openURLInNewTab(url)
```

Comment:
I notice we have this same code in `SettingsTableViewController` for the Help settings item. It'd be nice if we could create a `SettingsDelegate` that `TabTrayController` implemented, which we could call in these places instead of fiddling with the parent view hierarchy here. I think doing that would also allow us to use `TabManager.openNewTab()` inside of `TabTrayController` instead of having to pull out a BVC from the app delegate to call `openURLInNewTab`, which makes me cry a little inside.

Since we already have this code in our settings, though, we can do this cleanup in a follow-up.